c/secp256k1_blake160_sighash_all.c

```diff
@@ -5,16 +5,19 @@
  * 0. program name
  * 1. pubkey blake160 hash, blake2b hash of pubkey first 20 bytes, used to shield the real
  * pubkey in lock script.
- * 
+ *
  * Witness:
  * 2. pubkey, real pubkey used to identify token owner
  * 3. signature, signature used to present ownership
+ * 4. signature size
  */
 int main(int argc, char* argv[])
 {
-  if (argc != 4) {
+  uint64_t length = 0;
```

Comment:
Question: `verify_sighash_all`'s last arg is of type `size_t`. Is there any problem passing `uint64_t` to that?
Sorry if this seems silly as I've almost forgotten these differences between C types 😿.

On a second thought, we only use 64bit (for VM and others), so `size_t` is always unsigned long long, which is exactly `uint64_t`?

Reply:
`size_t` is defined to hold the biggest number that can be represented in the system namespace. Since we are using 64-bit system, casting it to `uint64_t` is safe.

Since we are using ilp64 model now, `size_t`, `uint64_t`, `unsigned long` and `unsigned long long` are all defined as 64 bit.

Reply:
Good to know :thumbup